satpy/readers/generic_image.py

```diff
@@ -102,12 +106,29 @@ def end_time(self):
 
     def get_dataset(self, key, info):
         """Get a dataset from the file."""
-        logger.debug("Reading %s.", key)
-        return self.file_content[key['name']]
+        ds_name = self.dataset_name if self.dataset_name else key['name']
```

Comment:
Just to make sure I understand, the idea here is that if a developer wants to, they could redefine the `read` method, set `dataset_name` to something else (or not at all) and then load more than just a basic image from an input dataset, right? Put another way, it gives people subclassing this file handler the flexibility to do more complex stuff?

Reply:
Yes, like the fake file handler in the test. Though I didn't use different names there.